File: app/src/main/java/com/chatserver/dd/chat/Cmd/Presenter/PresenterCmd.java
package com.chatserver.dd.chat.Cmd.Presenter;

import com.chatserver.dd.chat.Cmd.Model.IModelCmd;
import com.chatserver.dd.chat.Cmd.View.IViewCmd;
import com.chatserver.dd.chat.Cmd.View.ViewConsoleCmd;
import com.chatserver.dd.chat.Cmd.View.ViewStatusPlayers;

public class PresenterCmd implements IPresenterCmd, IModelCmd.OnFinishedListener, IModelCmd.OnFinishedListenerCmd {


    IViewCmd iViewCmd;
    IModelCmd iModelCmd;

    public PresenterCmd(IViewCmd iViewCmd, IModelCmd iModelCmd) {
        this.iViewCmd = iViewCmd;
        this.iModelCmd = iModelCmd;
    }

    @Override
    public void requestDataFromServer() {
        iViewCmd.showProgressBar();
        iModelCmd.getPlayersList(this);
    }

    @Override
    public void onExecuteCmd(String cmdCommand) {
        iModelCmd.onExecuteCmd(this, cmdCommand);
    }

    @Override
    public void onRefreshButtonClick() {
        if (iViewCmd != null) {

        }
        iModelCmd.getPlayersList(this);


    }

    //replace cmd Command
    @Override
    public void onSetCmdToEditText(String playerName) {

        //String to Array String
        String textInEdittext = iViewCmd.getEdittext();
        String[] ary = textInEdittext.split(" ");
        ary[0] = playerName;

        //array to String
        StringBuilder builder = new StringBuilder();
        for (String s : ary) {
            builder.append(s + " ");
        }
        String str = builder.toString();

        //set string to Edittext
        iViewCmd.setEdittext(str);
    }

    //replace playerName
    @Override
    public void onSetNameToEditText(String playerName) {
        String textInEdittext = iViewCmd.getEdittext();
        String[] ary = textInEdittext.split(" ");
        iViewCmd.setEdittext(ary[0] + " " + playerName);

    }


    @Override
    public void onFinishedPlayers(ViewStatusPlayers viewStatusPlayers) {

        iViewCmd.setDataToRecyclerView(viewStatusPlayers);

        iViewCmd.hideProgressBar();
        if (viewStatusPlayers.getData().getP().size() > 0) {
            iViewCmd.listIsNotEmpty();
        } else {
            iViewCmd.listIsEmpty();
        }
    }

    @Override
    public void onFailurePlayers(Throwable t) {
        iViewCmd.onResponseFailure(t);
    }

    @Override
    public void onFinishedCmd(ViewConsoleCmd viewConsoleCmd) {
        if (viewConsoleCmd != null) {
            iViewCmd.showToast(viewConsoleCmd.getStatus());
        }
    }

    @Override
    public void onFailureCmd(Throwable t) {
        iViewCmd.onResponseFailure(t);
    }


}

File: app/src/main/java/com/chatserver/dd/chat/Cmd/Presenter/PresenterCmd.kt
package com.chatserver.dd.chat.Cmd.Presenter

import com.chatserver.dd.chat.Cmd.Model.IModelCmd
import com.chatserver.dd.chat.Cmd.Model.IModelCmd.OnFinishedListenerCmd
import com.chatserver.dd.chat.Cmd.View.IViewCmd
import com.chatserver.dd.chat.Cmd.View.ViewConsoleCmd
import com.chatserver.dd.chat.Cmd.View.ViewStatusPlayers

class PresenterCmd(var iViewCmd: IViewCmd, var iModelCmd: IModelCmd) : IPresenterCmd, IModelCmd.OnFinishedListener, OnFinishedListenerCmd {
    override fun requestDataFromServer() {
        iViewCmd.showProgressBar()
        iModelCmd.getPlayersList(this)
    }

    override fun onExecuteCmd(cmdCommand: String) {
        iModelCmd.onExecuteCmd(this, cmdCommand)
    }

    override fun onRefreshButtonClick() {
        iModelCmd.getPlayersList(this)
    }

    //replace cmd Command
    override fun onSetCmdToEditText(playerName: String) {
        //String to Array String
        val textInEdittext = iViewCmd.edittext
        val ary = textInEdittext?.split(" ".toRegex())?.toTypedArray()
        ary?.set(0, playerName)
        //array to String
        val builder = StringBuilder()
        if (ary != null) {
            for (s in ary) {
                builder.append("$s ")
            }
        }
        val str = builder.toString()
        //set string to Edittext
        iViewCmd.edittext = str
    }

    //replace playerName
    override fun onSetNameToEditText(playerName: String) {
        val textInEdittext = iViewCmd.edittext
        val ary = textInEdittext?.split(" ".toRegex())?.toTypedArray()
        iViewCmd.edittext = (ary?.get(0) ?: "") + " " + playerName
        iViewCmd.edittext = "flags_add_user \"${playerName}\" \"t\" \"+7200\" \"${playerName}\" \"FreeVip\""
    }

    override fun onFinishedPlayers(viewStatusPlayers: ViewStatusPlayers) {
        val sorted = viewStatusPlayers.data.p.sortedByDescending {
            it.time
        }

        iViewCmd.setDataToRecyclerView(sorted)
        iViewCmd.hideProgressBar()
        if (viewStatusPlayers.data.p.size > 0) {
            iViewCmd.listIsNotEmpty()
        } else {
            iViewCmd.listIsEmpty()
        }
    }

    override fun onFailurePlayers(t: Throwable) {
        iViewCmd.onResponseFailure(t)
    }

    override fun onFinishedCmd(viewConsoleCmd: ViewConsoleCmd) {
        iViewCmd.showToast(viewConsoleCmd.status)
    }

    override fun onFailureCmd(t: Throwable) {
        iViewCmd.onResponseFailure(t)
    }
}